src/AutoRest.CSharp/MgmtTest/Models/MockTestCase.cs

```diff
@@ -79,7 +81,7 @@ private MgmtRestOperation GetRestOperationFromOperationId()
         /// <param name="requestPath"></param>
         /// <returns></returns>
         /// <exception cref="InvalidOperationException"></exception>
-        public IEnumerable<FormattableString> ComposeResourceIdentifierParameterValues(RequestPath requestPath)
+        public IEnumerable<FormattableString> ComposeResourceIdentifierParameterValues(RequestPath requestPath, bool generatingSample)
```

Comment:
I do not like the fact that we will have to check the literal values to determine whether this is a subscriptionId.
If we have to do something differently when it is a subscriptionId, how about (take either)
- instead of returning a FormattableString, we return a collection of a class when some more verbose information, we could add anything in that (for instance, `IsSubscriptionId`)
- if the extra action only is done to subscriptionId, since we already have that check inside this method, why do we do it here?